telemetry/request_factory.go

```diff
@@ -186,7 +186,7 @@ func NewSpanRequestFactory(options ...ClientOption) (RequestFactory, error) {
 		return nil, err
 	}
 
-	return &hashRequestFactory{requestFactory: f}, nil
+	return &eventRequestFactory{requestFactory: f}, nil
```

Comment:
This feels weird to me. The way we need to construct payloads now matches the `eventRequestFactory` strategy, but the naming convention seems weird. 

What about a `hashRequestFactory` implies a structure like: `[{ entry1, entry2 }]`
And what about a `eventRequestFactory` implies a structure like: `[ entry1, entry2 ]`

Reply:
Yes, I struggled with coming up with a decent name for the two options. If we update the code so that all batches can have an optional common block, then we can just have a single request factory. However, the current design keeps the common block as a separate PayloadEntry, so the idea between the two names comes from whether or not each PayloadEntry sent to the requestFactory would be combined in a hash (the hashRequestFactory), or if they would be combined in the way that the event API required (eventRequestFactory, which could have also been the array/slice request factory).

Reply:
Maybe `arrayRequestFactory` and `mapRequestFactory`?

That's really what's happening here...
- An `eventRequestFactory` (could become `arrayRequestFactory`) encodes each payloadEntry and wraps the result in `[ ... ]`. Its really assuming that each payload entry is an element in a json array.
- A `hashRequestFactory` (could become `mapRequestFactory`) encodes each payloadEntry as key / value pairs in the map of a single element array `[{ ... }]`